cpp/src/experimental/louvain.cuh

```diff
@@ -33,6 +32,12 @@
 #include <patterns/transform_reduce_e.cuh>
 #include <patterns/transform_reduce_v.cuh>
 
+#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ < 700
```

Comment:
I'm wondering if this empty `if` block deserves a comment. Alternatively, can you instead do the conditional include in the `if` block by checking for `__CUDA_ARCH__` >= 700 ? I'm also wondering what the expected behavior should be if `__CUDA_ARCH__` is not defined and what that means - should we still include static_map (which is what will happen here)?   (FWIW, I believe if `__CUDA_ARCH__` is not defined the preprocessor treats it like 0 in the expression, based on [this](https://gcc.gnu.org/onlinedocs/gcc-3.0.1/cpp_4.html#SEC39))

note: I just noticed the other conditional preprocessor directives for detecting Pascal are using the same expression, so maybe you're just trying to use the same line everywhere even if you want the negative, hence the empty if block?  That makes sense, I have no strong preference, but I'm still wondering what it means to not have `__CUDA_ARCH__` defined and what the behavior should be.

Reply:
I should add the FIXME comment block I use elsewhere and enhance it for the description here.

Reply:
Sounds good, thanks.  I'll mark this as resolved pending the enhanced FIXME to unblock the PR.